cylc/sphinx_ext/cylc_lang/__init__.py

```diff
@@ -181,19 +199,51 @@
 
          This resets it to the hardcoded default which is ``flow.cylc``.
 
+
+
 '''
 
-from cylc.sphinx_ext.cylc_lang.autodocumenters import (
-    CylcAutoDirective,
-    CylcAutoTypeDirective
-)
-from cylc.sphinx_ext.cylc_lang.domains import (
-    ParsecDomain,
-    CylcDomain,
-    CylcScopeDirective
-)
-from cylc.sphinx_ext.cylc_lang.lexers import CylcLexer, CylcGraphLexer
+rawdoc2 = """
+.. rst:directive:: .. auto-global-cylc:: source
+
+   Get a Cylc Global Configuration and render metadata fields.
+
+   If the optional source argument is give,
+   set ``CYLC_SITE_CONF_PATH`` to this value.
+
+   .. note::
+
+      If you have a user config this will still override the site
+      config!
+
+   .. rst-example::
+
+      .. auto-cylc-global:: {workflow_path}
+         :show:
```

Comment:
@oliver-sanders  : I'm wondering if this is too half-hearted a form of customization and whether I should change this to be:

* No args - makes some sensible assumptions.
* Show - User specifies what they want explicitly.

Reply:
Take a look at how other Sphinx autodocumenters do it and copy their approach.

https://github.com/search?q=repo%3Acylc%2Fcylc-doc%20%3Amembers%3A&type=code

Reply:
That looks like what I had in mind